File: lib/models/analyse.dart
class ResultatAnalyse {
  final String parametre;
  final String valeur;
  final String unite;
  final String statut; // 'normal' | 'bas' | 'eleve' | 'critique'

  ResultatAnalyse({
    required this.parametre,
    required this.valeur,
    required this.unite,
    required this.statut,
  });

  factory ResultatAnalyse.fromJson(Map<String, dynamic> json) {
    return ResultatAnalyse(
      parametre: json['parametre'] ?? '',
      valeur: json['valeur'] ?? '',
      unite: json['unite'] ?? '',
      statut: json['statut'] ?? 'normal',
    );
  }
}

class Analyse {
  final String id;
  final String type; // 'NFS', 'GE', 'Glycémie à jeun', etc.
  final List<ResultatAnalyse> resultats;
  final DateTime date;
  final String medecinNom;

  Analyse({
    required this.id,
    required this.type,
    required this.resultats,
    required this.date,
    required this.medecinNom,
  });

  factory Analyse.fromJson(Map<String, dynamic> json) {
    return Analyse(
      id: json['_id'] ?? '',
      type: json['type'] ?? '',
      resultats:
          (json['resultats'] as List?)
              ?.map((r) => ResultatAnalyse.fromJson(r as Map<String, dynamic>))
              .toList() ??
          [],
      date: DateTime.parse(json['date'] ?? DateTime.now().toIso8601String()),
      medecinNom: json['medecinNom'] ?? 'N/A',
    );
  }
}

File: backend/src/models/Analyse.js
const mongoose = require('mongoose');

const resultatSchema = new mongoose.Schema(
  {
    parametre: { type: String, required: true },
    valeur: { type: String, required: true },
    unite: { type: String, default: '' },
    statut: {
      type: String,
      enum: ['normal', 'bas', 'eleve', 'critique'],
      default: 'normal',
    },
  },
  { _id: false }
);

const analyseSchema = new mongoose.Schema(
  {
    patient: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Patient',
      required: true,
    },
    clinique: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Clinique',
      required: true,
    },
    medecin: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    type: { type: String, required: true },
    resultats: [resultatSchema],
    date: { type: Date, default: Date.now },
  },
  { timestamps: true }
);

const Analyse = mongoose.model('Analyse', analyseSchema);

module.exports = Analyse;
